runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -2442,7 +2442,7 @@ MM_CopyForwardScheme::createNextSplitArrayWorkUnit(MM_EnvironmentVLHGC *env, J9I
 				noEvacuation = isObjectInNoEvacuationRegions(env, (J9Object *) arrayPtr);
 			}
 
-			if (_abortInProgress || noEvacuation) {
+			if (_abortInProgress || noEvacuation || (abortFlagRaised() && currentSplitUnitOnly)) {
```

Comment:
'&& currentSplitUnitOnly' part does not make sense. If currently only, then we don't want to push the remainder (and indeed there is guard just after that).

As before, if we are adding abortFlagRaised (what seems right), we can probably drop _abortInProgress.